api/v1/iprangeclaim_types.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2024 Swisscom (Schweiz) AG.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// IpRangeClaimSpec defines the desired state of IpRangeClaim
+type IpRangeClaimSpec struct {
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:Format=cidr
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="Field 'parentPrefix' is immutable"
+	ParentPrefix string `json:"parentPrefix"`
+
+	//+kubebuilder:validation:Required
+	//+kubebuilder:validation:Minimum=2
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="Field 'size' is immutable"
+	Size int `json:"size,omitempty"`
```

Comment:
There is a current de-facto hard limit here which is at 50 (should be reproducible by trying to create a iprc with .spec.size 51 in an empty prefix) and a soft limit that might be even lower (depending on the fragmentation of IPs in the prefix). I assume this is due to pagination in the client. I propose to add validation Maximum=50 here and add a comment to this field.

Reply:
Added a validation rule for maximum value for size until pagination is added to the function searching an available iprange.

Reply:
Can you maybe make the comment a bit more verbose and more about the why (the what is covered in the validation). Something like:

"Size is the amount of consecutive IP Addresses you wish to reserve. Currently only sizes up to 50 are supported due to pagination of the NetBox API. In practice, this might be even lower depending on the fragmentation of the parent prefix."

Reply:
Updated the description with more information.